### PR TITLE
Some patches in prep for clustering

### PR DESCRIPTION
**Pull request by tych0**
Here's some random patches that add or abstract bits that we'll need for clustering.

### CODE REVIEW

**Comment by tych0** [comment]
Heh. Looks like I have to drop at least one patch since it's not actually used in the current code :\

**Comment by stgraber** [comment]
jenkins: test this please

File: lxd/daemon.go [hunk 1]
```diff
@@ -748,7 +748,7 @@ func (d *Daemon) Init() error {
 	d.lxcpath = shared.VarPath("containers")
 
 	/* Make sure all our directories are available */
-	if err := os.MkdirAll(shared.CachePath(), 0700); err != nil {
+	if err := os.MkdirAll(shared.CachePath(), 0711); err != nil {
```
Comment (stgraber) [inline lxd/daemon.go]: Hmm, why would all users have to be allowed to traverse to /var/cache/lxd?

Note that above is shared.CachePath() not shared.VarPath()

File: client.go [hunk 2]
```diff
@@ -1676,10 +1675,10 @@ func (c *Client) ServerStatus() (*shared.ServerState, error) {
 
 	// Fill in certificate fingerprint if not provided
 	if ss.Environment.CertificateFingerprint == "" && ss.Environment.Certificate != "" {
-		pemCertificate, _ := pem.Decode([]byte(ss.Environment.Certificate))
-		if pemCertificate != nil {
-			digest := sha256.Sum256(pemCertificate.Bytes)
-			ss.Environment.CertificateFingerprint = fmt.Sprintf("%x", digest)
+		var err error
+		ss.Environment.CertificateFingerprint, err = shared.CertFingerprint(ss.Environment.Certificate)
```
Comment (stgraber) [inline client.go]: Same should probably be done to the other two functions that hash certificates (lxd/certificates.go and lxc/remote.go).

File: lxd/daemon.go (hunk 1 above)
Comment (tych0) [inline lxd/daemon.go]: No reason, it's just the first mkdir that we call. I can do one for VarPath that's 711 and then do a separate one for CachePath that's 700.

File: client.go (hunk 2 above)
Comment (tych0) [inline client.go]: Will do.

File: lxd/daemon.go (hunk 1 above)
Comment (tych0) [inline lxd/daemon.go]: Oh, then yes, this is totally wrong in that case :). Anyway, I pushed a patch which does it for VarPath.

**Pull request merged**